lib/plottr_components/src/components/project/BookDialog.js

```diff
@@ -24,38 +30,67 @@ const BookDialogConnector = (connector) => {
       this.themeRef = null
     }
 
-    saveEdit = () => {
-      const { book } = this.props
+    saveEdit = (saveAndOpen) => {
+      const { book, books } = this.props
 
       let title = this.titleRef.value
       let premise = this.premiseRef.value
       let genre = this.genreRef.value
       let theme = this.themeRef.value
-      this.props.actions.editBook(book.id, { title, premise, genre, theme })
-      this.props.cancel()
+      const newBookId = objectId(books.allIds)
+
+      if (!book) {
+        this.props.actions.addBook(newBookId, { title, premise, genre, theme })
```

Comment:
Please defer as much logic as you can to `pltr`.
In this instance, you've computed the new book id in the component, but we could have done so in the reducer instead.
If there's no logic required to **use** `pltr`, then our components become cleaner and will likely have fewer bugs.

To that end, please:
 - remove `newBookId` from `addBook`, and
 - spread its other arguments out of that object.  i.e. please change the signature of add book to: `addBook(title, premise, genre, theme)`.  This will make it trivial for others to know what to supply.   If you keep it as an object, then whoever uses `pltr` has to research how it was used in other projects.
 
Finally, if possible, please change the signature of `editBook` to `editBook(bookId, title, premise, genre, theme)`.
If `editBook` is used in more places than just here, then please document the change in the readme of `pltr` in a new section like so:
```
# Breaking Changes
## <today's date>
 - `editBook(bookId, {title, premise, genre, theme}) -> editBook(bookId, title, premise, genre, theme)`
```